src/webgpu/api/validation/render_pipeline/depth_stencil_state.spec.ts

```diff
@@ -32,6 +32,53 @@ g.test('format')
     t.doCreateRenderPipelineTest(isAsync, !!info.depth || !!info.stencil, descriptor);
   });
 
+g.test('depthCompare_optional')
+  .desc(
+    `The depthCompare in depthStencilState is optional for stencil-only formats but required for formats with a depth.`
+  )
+  .params(u =>
```

Comment:
IMHO these should be subcases, I think paramsSubCaseOnly is a thing?

Reply:
As communicated offline, `paramsSubcasesOnly` and `.beforeAllSubcases(t => { const { format } = t.params; ...` don't seem to work together.

Reply:
@Kangz Using `beginSubcases()` works better! Is this what you had in mind?
<img width="1840" alt="image" src="https://github.com/gpuweb/cts/assets/634478/fcafa630-0b4b-4018-b48e-a835bdd8331d">

Reply:
> As communicated offline, `paramsSubcasesOnly` and `.beforeAllSubcases(t => { const { format } = t.params; ...` don't seem to work together.

Interesting, I would have thought they did. Glad you found a workaround though!